ras-arm-handler.c

```diff
@@ -78,6 +99,48 @@ int ras_arm_event_handler(struct trace_seq *s,
 	ev.psci_state = val;
 	trace_seq_printf(s, "\n psci_state: %d", ev.psci_state);
 
+	if (pevent_get_field_val(s, event, "pei_len", record, &val, 1) < 0)
+		return -1;
+	ev.pei_len = val;
+	trace_seq_printf(s, "\n ARM Processor Err Info data len: %d\n",
+			 ev.pei_len);
+
+	ev.pei_error = pevent_get_field_raw(s, event, "buf", record, &len, 1);
+	if (!ev.pei_error)
+		return -1;
+	display_raw_data(s, ev.pei_error, ev.pei_len);
+
+	if (pevent_get_field_val(s, event, "ctx_len", record, &val, 1) < 0)
+		return -1;
+	ev.ctx_len = val;
+	trace_seq_printf(s, "\n ARM Processor Err Context Info data len: %d\n",
+			 ev.ctx_len);
+
+	ev.ctx_error = pevent_get_field_raw(s, event, "buf1", record, &len, 1);
+	if (!ev.ctx_error)
+		return -1;
+	display_raw_data(s, ev.ctx_error, ev.ctx_len);
+
+	if (pevent_get_field_val(s, event, "oem_len", record, &val, 1) < 0)
+		return -1;
+	ev.oem_len = val;
+	trace_seq_printf(s, "\n Vendor Specific Err Info data len: %d\n",
+			 ev.oem_len);
+
+	ev.vsei_error = pevent_get_field_raw(s, event, "buf2", record, &len, 1);
+	if (!ev.vsei_error)
+		return -1;
```

Comment:
Hmm... are the above fields present on ARM CPUs from all vendors, and for all CPU models? If not, then some check is needed to only enable them when the right CPU models are available.

Reply:
This is for all ARM CPUs. The original arm_event trace code only traces out ARM processor error information data. According to UEFI_2_8_A_Feb14 specification chapter N2.4.4, the ARM processor error section includes several
ARM processor error information, several ARM processor context information and several vendor specific error
information structures. 

the ARM part is added in UEFI specification 2.6, the ARM Processor Error Section format is same and includes ARM content and vendor specific data. Also code will check the length of data by reading ARM Processor Error Section record, code won't print anything if it is zero.